libs/repositories/src/UsdRepository/UsdRepositoryRedis.ts

```diff
@@ -0,0 +1,119 @@
+import { injectable } from 'inversify';
+import {
+  PricePoint,
+  PriceStrategy,
+  UsdRepository,
+  deserializePricePoints,
+  serializePricePoints,
+} from './UsdRepository';
+import { SupportedChainId } from '../types';
+import IORedis from 'ioredis';
+
+const DEFAULT_CACHE_TIME_SECONDS = 120; // 2min cache time by default for USD prices
+const NULL_VALUE = 'null';
+
+@injectable()
+export class UsdRepositoryRedis implements UsdRepository {
+  private baseCacheKey: string;
+
+  constructor(
+    private proxy: UsdRepository,
+    private redisClient: IORedis,
+    private cacheName: string,
+    private cacheTimeSeconds: number = DEFAULT_CACHE_TIME_SECONDS
+  ) {
+    this.baseCacheKey = `repos:${this.cacheName}`;
```

Comment:
Nitpick: can it be `repo` instead? It's meant as a single repository per cache name, not multiple under the same key.

```suggestion
    this.baseCacheKey = `repo:${this.cacheName}`;
```

Reply:
Redis uses colon notation to group keys. Clients will show it as a tree view. So you will see `repos` and then inside you will see `usdCoingecko`, `usdCow` etc

This is why I used the plural. I did the same for caching the `requests`